algorithms/CPlusPlus/Arrays/search-insert-position.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+Description: Given a sorted array of distinct integers and a target value, 
+return the index if the target is found. If not, return the index where it
+would be if it were inserted in order.
+
+Approach: Using Binary search to solve in least time complexity.
+
+Time Complexity: O(log n)
+*/
+
+#include <iostream>
+#include <vector>
+using namespace std;
+
+//function 
+int searchInsert(vector<int>& nums, int target) {
+    //starting index
+    int start=0;
+    //ending index
+    int end=nums.size()-1;
+    while(start<=end){
+        //calculating mid element
+        int mid = start + (end-start)/2;
+        //if target found, return the index
+        if(nums[mid]==target){
+            return mid;
+        }
+        if(nums[mid]>target){
+             end=mid-1;
+        }
+        else{
+            start=mid+1;
+       }
+    }
+    //returning end+1 so that we can get the index where the target element can be inserted
+    return end+1;
+        
+}
+
+//main starts
+int main() {
+    cout << "Enter number of elements in the array\n";
+    int n;
+    cin>>n;
+    cout<<"Enter the array elements\n";
+    vector<int> nums(n);
+    for(int i=0;i<n;i++){
+        cin>>nums[i];
+    }
+    cout<<"Enter the target element\n";
+    int target;
+    cin>>target;
+    cout<<"The target element can be inserted at: "<<searchInsert(nums,target);
```

Comment:
The text here can be misleading as it prints `The target element can be inserted at: (some_index)` regardless of whether the target is present or not.

You don't want to print this if the target is present, in such cases a valid statement will be 
`The location of the target is (some_index)`.

My suggestion is to have a valid statement for both the cases(when the target is present & when the target is not present), you can make changes in the function itself accordingly.